### ISSUE TITLE
`Style/EvalWithLocation` could auto-correct

### ISSUE THREAD

**Issue by marcandre** [rubocop-hq/rubocop#9403]
The cop `Style/EvalWithLocation` could autocorrect:

```ruby
example.class_eval <<~RUBY
  etc...
RUBY
# to
example.class_eval <<~RUBY, __FILE__, __LINE__ + 1
  etc...
RUBY
```

**Comment by caalberts** [comment]
For `#eval`, the current cop is suggesting to pass `__FILE__` and `__LINE__`, but there needs to be  a binding as second argument.

```rb
  it 'registers an offense when using `#eval` without any arguments' do
    expect_offense(<<~RUBY)
      eval <<-CODE
      ^^^^^^^^^^^^ Pass `__FILE__` and `__LINE__` to `eval` method, as they are used by backtraces.
        do_something
      CODE
    RUBY
```

Without binding, there would be an error:

```rb
eval(<<~CODE, __FILE__, __LINE__ + 1)
  raise 'test error in eval'
CODE

# Traceback (most recent call last):
#        1: from sample.rb:3:in `<main>'
# sample.rb:3:in `eval': no implicit conversion of Integer into String (TypeError)
```

I think this should be fixed first before adding auto-correct.

**Comment by dvandersluis** [comment]
I opened #9409 to fix the offense messages for this cop.

**Comment by ceteece** [comment]
I can take a stab at this issue this weekend if no one else has gotten a start on it.

**Comment by dvandersluis** [comment]
@ceteece go for it! Just be aware there are two PRs open that change the cop (particularly #9411 which refactors some of the code).

**Comment by ceteece** [comment]
Sounds good, I'll wait for those to get merged before submitting a PR.
